obsproject/obs-studio
# obs-text, text-freetype2: Add Enable Antialiasing option

### Description


This PR adds the option to disable antialiasing on font rendering. 

✅ Antialiasing is **Enabled** by default (the current behaviour).

### obs-text
![image](https://user-images.githubusercontent.com/8297838/80224877-71c65700-869e-11ea-9fcb-1f9d27de01ff.png)

The added option in the UI:
![image](https://user-images.githubusercontent.com/8297838/80225845-c0c0bc00-869f-11ea-85ab-24f9dd280127.png)

A scene with various text-gdi sources:
Before:
![image](https://user-images.githubusercontent.com/8297838/80225349-0e88f480-869f-11ea-8f21-d722af4a85fd.png)

After:
![image](https://user-images.githubusercontent.com/8297838/80225586-60ca1580-869f-11ea-814e-ba9407f4ae3f.png)

A scene with a mixture of enabled and disabled:
![image](https://user-images.githubusercontent.com/8297838/80225690-80f9d480-869f-11ea-8552-7ed6429a9afe.png)

### text-freetype2
The added option in the UI:
![image](https://user-images.githubusercontent.com/8297838/80602291-09eb8400-8a83-11ea-8e55-1267adf45eb3.png)

A scene with various text-freetype2 sources:
Before:
![image](https://user-images.githubusercontent.com/8297838/80602163-e1fc2080-8a82-11ea-9181-5e18ab3302f3.png)

After:
![image](https://user-images.githubusercontent.com/8297838/80602239-f93b0e00-8a82-11ea-813f-9de03a22ba36.png)

A scene with a mixture of enabled and disabled:
![image](https://user-images.githubusercontent.com/8297838/80602000-aeb99180-8a82-11ea-8040-72f97da3823b.png)

### Motivation and Context



Some fonts are styled in such a way that antialiasing them makes them unreadable or look undesirable, these being mainly pixel fonts.

With antialiasing disabled, pixel fonts become clear and easy to read.

### How Has This Been Tested?



Tested with both regular and pixel fonts, enabling and disabling to ensure that live changes are made.

- Add a text-gdi source.
- Disable antialiasing.
- Restart OBS to ensure setting persists.
- Enable antialiasing.
- Restart OBS to ensure setting persists.

Create a scene with various combinations of text off and on.

These steps have been repeated now with freetype2.

### Types of changes







- New feature (non-breaking change which adds functionality)

### Checklist:
